### PR TITLE
Tidy search filter code

### PR DESCRIPTION
**Pull request by calherries**
WIP. Inches us closer towards [search modularization](https://www.notion.so/metabase/Search-Modularization-part-1-7f3ed9b914a5455fbe311d25af565daa).

### CODE REVIEW

File: src/metabase/search/config.clj [hunk 1]
```diff
@@ -195,25 +205,11 @@
...
 
-(defmethod searchable-columns-for-model "action"
-  [_]
-  [:name
-   :dataset_query
-   :description])
-
-(defmethod searchable-columns-for-model "card"
+(doseq [model ["card" "dataset" "action" "metric"]]
+ (defmethod searchable-columns-for-model model
```
Comment (piranha) [inline src/metabase/search/config.clj:209]: Those loops look worse to me than previous code. I think making it more dense (lesser amount of newlines) is fine, but searchability of code with defmethod in a loop is not great.

File: src/metabase/search/filter.clj [hunk 2]
```diff
@@ -210,41 +145,56 @@
     (str/capitalize model)))
 
 (doseq [model ["dashboard" "card" "dataset" "metric"]]
```
Comment (piranha) [inline src/metabase/search/filter.clj:147]: Oh, they are everywhere already, disregard my comment

I think have much less code to support search is one of the goals we should have. 😁

File: src/metabase/search/config.clj (hunk 1 above)
Comment (calherries) [reply src/metabase/search/config.clj:209]: Good point, I hadn't considered the benefit of being able to search for a given model. Updated 
80f42c7599cd60976808744a30fe4b5cc72a8236

Comment (calherries) [reply src/metabase/search/config.clj]: (this code is going very soon anyway, but its still nice to refactor easier-to-read code)

File: src/metabase/search/filter.clj (hunk 2 above)
Comment (calherries) [reply src/metabase/search/filter.clj:147]: Still your comment had merit in that instance, but yes, not everywhere

**Review by sansolovyov — APPROVED** [review]
🥁

File: src/metabase/search/filter.clj [hunk 3]
```diff
@@ -13,149 +13,73 @@
...
-                              (remove #{:dataset_query})
-
-                              true
-                              (map #(search.config/column-with-model-alias model %)))
-           wildcarded-token (->> (search.util/normalize query)
-                                 search.util/tokenize
-                                 (map search.util/wildcard-match))]
-       (cond
-         (and (= model "indexed-entity") (sandboxed-or-impersonated-user?))
-         [:= 0 1]
```
Comment (calherries) [inline src/metabase/search/filter.clj:94]: I moved this to `search-query-for-model` here: https://github.com/metabase/metabase/pull/43715/files#diff-a4641d5a87a4c712c4d2210a0dcac41915018fb6c2322413235313f283c61e35R358

File: src/metabase/search/filter.clj [hunk 4]
```diff
@@ -13,149 +13,73 @@
...
-(defn- sandboxed-or-impersonated-user? []
-  ;; TODO FIXME -- search actually currently still requires [[metabase.api.common/*current-user*]] to be bound,
-  ;; because [[metabase.public-settings.premium-features/sandboxed-or-impersonated-user?]] requires it to be bound.
-  ;; Since it's part of the search context it would be nice if we could run search without having to bind that stuff at
-  ;; all.
-  (assert @@(requiring-resolve 'metabase.api.common/*current-user*)
-          "metabase.api.common/*current-user* must be bound in order to use search for an indexed entity")
-  (premium-features/sandboxed-or-impersonated-user?))
-
-(mu/defn ^:private search-string-clause-for-model
```
Comment (calherries) [inline src/metabase/search/filter.clj:76]: Excluding the `"indexed-entity"` stuff, this logic is now added to each query here: https://github.com/metabase/metabase/blob/188649a8a757e4830c029676e45437bfac1b71c7/src/metabase/search/impl.clj#L126

File: src/metabase/search/filter.clj [hunk 5]
```diff
@@ -13,149 +13,73 @@
...
-           wildcarded-token (->> (search.util/normalize query)
-                                 search.util/tokenize
-                                 (map search.util/wildcard-match))]
-       (cond
-         (and (= model "indexed-entity") (sandboxed-or-impersonated-user?))
-         [:= 0 1]
-
-         (and (#{"card" "dataset"} model) (= column (search.config/column-with-model-alias model :dataset_query)))
-         [:and
-          [:= (search.config/column-with-model-alias model :query_type) "native"]
```
Comment (calherries) [inline src/metabase/search/filter.clj:98]: Filtering for native queries is implemented here https://github.com/metabase/metabase/blob/188649a8a757e4830c029676e45437bfac1b71c7/src/metabase/search/impl.clj#L255-L256

**Comment by calherries** [comment]
closing since this is admin-webapp's now

**Pull request closed without merging**